frontend/packages/client/src/hooks/useUserRoleOnCommunity.js

```diff
@@ -1,27 +1,35 @@
+import { useCallback, useEffect, useState } from "react";
+import { isEmptyArray } from "utils";
 import useUserCommunities from "./useUserCommunities";
 
 export default function useUserRoleOnCommunity({
   addr,
   communityId,
   roles = [],
 } = {}) {
-  const { data: communityUser, loading } = useUserCommunities({
-    addr,
+
+  const [hasRole, setHasRole] = useState();
+  const { data: userCommunities, loading } = useUserCommunities({
+    addr
   });
 
-  if (
-    addr === null ||
-    (Array.isArray(communityUser) && communityUser.length === 0)
-  ) {
-    return false;
-  }
-  if (loading || roles.length === 0) {
-    return null;
-  }
-  const rolesInCommunity =
-    communityUser
-      ?.filter((datum) => datum.id.toString() === communityId.toString())
-      .map((community) => community?.membershipType) ?? [];
+  const getUserRole = useCallback(() => {
+    if (loading || isEmptyArray(roles) || !userCommunities) {
+      return setHasRole(null);
+    } else if (isEmptyArray(userCommunities)) {
+      return setHasRole(false);
+    } else {
+      const rolesInCommunity = userCommunities
+        .filter(({ id }) => id.toString() === communityId.toString())
+        .map(({ membershipType }) => membershipType);
+      const included = roles.every((role) => rolesInCommunity.includes(role));
+      setHasRole(included);
+    }
+  }, [communityId, loading, roles, userCommunities]);
+
+  useEffect(() => {
+    getUserRole();
+  }, [getUserRole]);
 
-  return roles.every((role) => rolesInCommunity.includes(role));
+  return hasRole;
```

Comment:
I think reusing the old code and moving the loading variable should work, then a state and a useEffect wound't be necessary:

```
 const { data: userCommunities, loading } = useUserCommunities({
    addr,
    count: 100,
  })
  if (loading) {
    return null;
  }
  if (
    addr === null ||
    (Array.isArray(userCommunities) && userCommunities.length === 0 && !loading)
  ) {
    return false;
  }

  const rolesInCommunity =
    userCommunities
      ?.filter((datum) => datum.id.toString() === communityId.toString())
      .map((community) => community?.membershipType) ?? [];
  return roles.every((role) => rolesInCommunity.includes(role));
```